bazel/bazel.sh

```diff
@@ -23,7 +23,8 @@ n=$(($#-$i+1))
 # Insert our flag at index $i out of $n. Quoting is all required for expansion.
 set -- "${@:1:$((i-1))}" "$BUILD_EVENT_FILE_FLAG" "${@:$i:$n}"
 
-/builder/bazel "$@"
```

Comment:
Why not just

```
/builder/bazel "$@"
EXIT_CODE=$?
```

(Sorry if there's some obscure bashism that makes the `&&` and `||` preferable, it could use a comment if so)

Reply:
Good question, the reason why is subtle. The whole script is run with `set -e` at the top, which ensures that if *any* pipeline in the script fails, the whole script fails. (I actually picked this idiom up from reading Docker tutorials - it's not totally obscure). It's a safety measure for scripts where one can assume any failure is a total failure.

In a script with the `-e` option enabled, there are two ways to deal with a single pipeline that might fail in the middle of your script. 

1. Temporarily disable the option:
```bash
set +e
/builder/bazel "$@"
EXIT_CODE=$?
set -e
```
1. Change the pipeline to succeed, minimally by adding a short-circuiting logical or expression (`|| true`):
```bash
EXIT_CODE=0
/builder/bazel "$@" || EXIT_CODE=$?
```

I added a comment and switched to the last code snippet I've included here, though I'm fine with any formulation. I can even drop the `set -e` entirely, though you might not like the extra error checking that change would introduce.

Reply:
Thanks that makes perfect sense.